Release version 7.5.0

sirosen opened the pull request:
These changes were applied via `towncrier build --version v7.5.0`.

<!--- Describe the changes here. --->

##### Contributor checklist

- [ ] Included tests for the changes.
- [ ] A change note is created in `changelog.d/` (see [`changelog.d/README.md`](https://github.com/jazzband/pip-tools/blob/main/changelog.d/#readme) for instructions) or the PR text says "no changelog needed".

##### Maintainer checklist

- [ ] If no changelog is needed, apply the `skip-changelog` label.
- [ ] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

webknjaz commented:
@sirosen I've been using a bunch of rejexes + `pandoc` to get GFM:

* https://github.com/ansible/pylibssh/blob/bf0528dcc6787d033d697dffbfd7dbab8e5e5814/packaging/pep517_backend/_transformers.py#L37-L118
* https://github.com/ansible/pylibssh/blob/bf0528dcc6787d033d697dffbfd7dbab8e5e5814/.github/workflows/ci-cd.yml#L537-L572

It'd be cool to generalize the solution. I know that `hatchling` has a place for hooks that can do such transformations nicely.

webknjaz commented:
Also, it doesn't have to be perfect — you can adjust the contents afterwards.

sirosen commented:
I unfortunately got pulled into some meetings so I'm delayed on this. I'll get back to it quite soon, within an hour or two.

sirosen commented:
I've just published it and will watch CI now.

Here's what I used for my first pass on the changelog:
<details><summary>changelog2md.py</summary>

```python
#!/usr/bin/env python
"""
Extract a changelog section from the full changelog contents, convert MyST
markdown and sphinx-issues syntaxes to GitHub-flavored Markdown.

Defaults to selecting the most recent (topmost) changelog section.
Can alternatively provide output for a specific version with `--target`.
e.g.

    ./changelog2md.py --target 3.20.0
"""
from __future__ import annotations

import argparse
import itertools
import pathlib
import re
import typing as t

REPO_ROOT = pathlib.Path(__file__).parent
CHANGELOG_PATH = REPO_ROOT / "CHANGELOG.md"

CHANGELOG_HEADER_PATTERN = re.compile(r"^## v(\d+\.\d+\.\d+)$", re.MULTILINE)

SPHINX_ISSUES_PR_PATTERN = re.compile(r"{pr}`(\d+)`")
SPHINX_ISSUES_ISSUE_PATTERN = re.compile(r"{issue}`(\d+)`")
SPHINX_ISSUES_USER_PATTERN = re.compile(r"{user}`([^`]+)`")


def iter_changelog_chunks(changelog_content: str) -> t.Iterator[tuple[str, str]]:
    # first one precedes the first changelog, slice it off
    chunks = CHANGELOG_HEADER_PATTERN.split(changelog_content)[1:]
    yield from itertools.pairwise(chunks)


def get_last_changelog(changelog_content: str) -> tuple[str, list[str]]:
    version, latest_changes = next(iter_changelog_chunks(changelog_content))
    lines = latest_changes.strip().splitlines()
    return version, lines


def get_changelog_section(target: str, changelog_content: str) -> tuple[str, list[str]]:
    for version, section in iter_changelog_chunks(changelog_content):
        if version == target:
            return version, section.strip().splitlines()
    raise LookupError(f"section for version={target} not found")


def convert_rst_to_md(version: str, lines: t.Iterable[str]) -> t.Iterator[str]:
    yield f"## v{version}"
    yield ""
    for line in lines:
        updated = line
        updated = SPHINX_ISSUES_PR_PATTERN.sub(r"#\1", updated)
        updated = SPHINX_ISSUES_ISSUE_PATTERN.sub(r"#\1", updated)
        updated = SPHINX_ISSUES_USER_PATTERN.sub(r"@\1", updated)
        updated = updated.replace("``", "`")
        yield updated


def main():
    parser = argparse.ArgumentParser(
        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
    )
    parser.add_argument(
        "--target", "-t", help="A target version to use. Defaults to latest."
    )
    args = parser.parse_args()

    full_changelog = CHANGELOG_PATH.read_text()

    if args.target:
        version, changelog_section = get_changelog_section(args.target, full_changelog)
    else:
        version, changelog_section = get_last_changelog(full_changelog)

    for line in convert_rst_to_md(version, changelog_section):
        print(line)


if __name__ == "__main__":
    main()
```

</details>

My previous version was for rst->md, hence the name (which doesn't quite fit here).
It could probably use some improvement (e.g., doesn't handle commit references) but it works for what we had in this release.

sirosen commented:
Hm. One little thing jumps out right away:
<img width="234" height="92" alt="image" src="https://github.com/user-attachments/assets/b0b644ac-7242-42a1-afc5-f0a2d6c13b71" />

Looks like we have some naming for the jobs which assumes a tag without `v` as a prefix. Should be easy to fix.
The build is still working properly, `setuptools-scm` is getting a version of 7.5.0.

webknjaz commented:
Wow, this is weird... I'm pretty sure @jezdez added me to that team but I'm not in it. Will have to ask again, I suppose.
